lib/auth/auth.js

```diff
@@ -26,120 +30,151 @@ auth.setAuthHandler = handler => {
     if (authHeader) {
-        log.trace('authorization header', { authHeader });
-        // TODO: Check for security token header to
-        // handle temporary security credentials
+        method = 'headers';
+        // TODO: Check for security token header to handle temporary security
+        // credentials
         if (authHeader.startsWith('AWS ')) {
-            log.trace('authenticating request with auth v2 using headers');
-            return authV2.headerAuthCheck.check(request, log, data);
+            version = 'v2';
         } else if (authHeader.startsWith('AWS4')) {
-            log.trace('authenticating request with Auth V4 using headers');
-            return authV4.headerAuthCheck.check(request, log, data,
-                awsService);
+            version = 'v4';
+        } else {
+            log.trace('missing authorization security header',
+                      { header: authHeader });
+            return { err: errors.MissingSecurityHeader };
         }
-        log.debug('missing authorization security header');
-        return { err: errors.MissingSecurityHeader };
     } else if (data.Signature) {
-        // Check whether signature is in query string
-        log.trace('authenticating request with auth v2 using query string');
-        return authV2.queryAuthCheck.check(request, log, data);
+        method = 'query';
+        version = 'v2';
     } else if (data['X-Amz-Algorithm']) {
-        log.trace('authenticating request with Auth v4 using query string');
-        return authV4.queryAuthCheck.check(request, log, data);
+        method = 'query';
+        version = 'v4';
+    }
+
+    // Here, either both values are set, or none is set
+    if (version !== null && method !== null) {
+        if (!checkFunctions[version] || !checkFunctions[version][method]) {
+            log.trace('invalid auth version or method', { version, method });
+            return { err: errors.NotImplemented };
+        }
+        log.trace('identified auth method', { version, method });
+        return checkFunctions[version][method](request, log, data, awsService);
     }
-    // If no auth information is provided in request, then
-    // user is part of 'All Users Group' so send back this
-    // group as the canonicalID
-    log.debug('No authentication provided. User identified as public');
-    const authInfo = new AuthInfo({ canonicalID: constants.publicId });
-    return { err: null, data: authInfo };
-};
 
-auth.doAuth = (request, log, cb, awsService, requestContext) => {
-    const res = auth.check(request, log, awsService, request.query);
+    // no auth info identified
+    log.debug('assuming public user');
+    return { err: null, params: publicUserInfo };
+}
+
+function doAuth(request, log, cb, awsService, requestContext) {
+    const res = extractParams(request, log, awsService, request.query);
     requestContext.setAuthType(res.authType);
```

Comment:
indeed. Missed that in my rebase. thanks, I'm guessing we'll see a few other similar mistakes.